src/pages/Signup/Signup.js

```diff
@@ -0,0 +1,310 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { RiArrowRightSLine } from 'react-icons/ri';
+import './Signup.scss';
+
+class Signup extends Component {
+  state = {
+    form: {
+      account: '',
+      password: '',
+      pwCheck: '',
+      name: '',
+      year: '',
+      month: '',
+      day: '',
+      email: '',
+      emailDot: '',
+      phoneNum1: '',
+      phoneNum2: '',
+      phoneNum3: '',
+    },
+    phoneNum: '',
+    date: '',
+    email: '',
+  };
+
+  handleOnchange = e => {
+    const { name, value } = e.target;
+    const { form } = this.state;
+    this.setState(
+      {
+        ...this.state,
+        form: {
+          ...this.state.form,
+          [name]: value,
+        },
+      },
+      () => {
+        this.setState({
+          ...this.state.phoneNum,
+          ...this.state.date,
+          phoneNum:
+            form.phoneNum1 + '-' + form.phoneNum2 + '-' + form.phoneNum3,
+          date: form.year + '-' + form.month + '-' + form.day,
+          email: form.email + '@' + form.emailDot,
+        });
```

Comment:
state 값을 콘솔 찍어보면 알겠지만, 이렇게 되면 state 안에 외도치 않은 값들이 들어가게 됩니다. 

지금 this.state.phoneNum 는 숫자 값인데, 이것을 스프레드 하면 다음과 같은 의미를 가집니다. `{ 123 }` 이것은 어떤 것을 의미하냐면, `{ '123' : 123 }` 과 같이 123이라는 키값에 숫자 123 을 담으라는 것이죠. 

그리고 setState 는 병합한다라는 것을 생각하시면 아예 필요없는 부분이겠네요!
```suggestion
this.setState({
  phoneNum: form.phoneNum1 + '-' + form.phoneNum2 + '-' + form.phoneNum3,
  date: form.year + '-' + form.month + '-' + form.day,
  email: form.email + '@' + form.emailDot,
 })
```

Reply:
네 이부분 console에서 확인했습니다. 제가 원하는 값이 아니라 이상한 값들이 들어가고 있었습니다!!